### PR TITLE
use `sig_realloc` in method `OP_make_set`

### PR DESCRIPTION
**Pull request by dcoudert**
Method `OP_make_set` was introduced in #38692 in file `src/sage/groups/perm_gps/partn_ref/data_structures.pxd` to add the method `make_set` to `DisjointSet` (i.e., the ability to add a new element to a disjoint set).

We improve the method by using `sig_realloc` rather than a combination of `sig_malloc` and `memcpy`. This requires some changes in methods `OP_new` and `OP_dealloc`.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/groups/perm_gps/partn_ref/data_structures.pxd [hunk 1]
```diff
@@ -138,38 +141,7 @@ cdef inline void OP_join(OrbitPartition *OP, int m, int n) noexcept:
...
-    new_mcr[n] = n
-    new_size[n] = 1
-
-    sig_free(OP.parent)
-
-    OP.parent = new_parent
-    OP.rank = new_rank
-    OP.mcr = new_mcr
-    OP.size = new_size
+cdef void OP_make_set(OrbitPartition *OP) noexcept
```
Comment (gmou3) [inline src/sage/groups/perm_gps/partn_ref/data_structures.pxd:144]:
```suggestion
cdef inline void OP_make_set(OrbitPartition *OP) noexcept
```

File: src/sage/groups/perm_gps/partn_ref/data_structures.pyx [hunk 2]
```diff
@@ -78,6 +85,32 @@ cdef OP_string(OrbitPartition *OP):
     return s
 
 
+cdef inline void OP_make_set(OrbitPartition *OP) noexcept:
+    """
+    Increase the degree of the input partition by one.
```
Comment (gmou3) [inline src/sage/groups/perm_gps/partn_ref/data_structures.pyx:92]:
```suggestion
    Increase the degree of the input partition by one.

```

**Review by gmou3 — COMMENTED** [review]
Overall looks cleaner to me.

File: src/sage/groups/perm_gps/partn_ref/data_structures.pxd (hunk 1 above)
Comment (dcoudert) [reply src/sage/groups/perm_gps/partn_ref/data_structures.pxd:144]: done

File: src/sage/groups/perm_gps/partn_ref/data_structures.pyx (hunk 2 above)
Comment (dcoudert) [reply src/sage/groups/perm_gps/partn_ref/data_structures.pyx:92]: done

File: src/sage/groups/perm_gps/partn_ref/data_structures.pxd (hunk 1 above)
Comment (dcoudert) [reply src/sage/groups/perm_gps/partn_ref/data_structures.pxd:144]: Actually, I have a compilation warning
```
[sagelib-10.6.beta0] [spkg-install]     warning: sage/groups/perm_gps/partn_ref/data_structures.pxd:144:28: Declarations should not be declared inline.
```
So I will revert this change.

Comment (gmou3) [reply src/sage/groups/perm_gps/partn_ref/data_structures.pxd:144]: Ok! Why is it that you moved the function definition to the `pyx` file?

Comment (dcoudert) [reply src/sage/groups/perm_gps/partn_ref/data_structures.pxd:144]: May be I'm mistaken, but isn't it the normal way to proceed, as in C with .h (headers) and .c (contents) files ?
I can of course move it back if it is better.

**Review by gmou3 — APPROVED** [review]

**Pull request merged**